src/main/java/ch/uzh/ifi/hase/soprafs23/controller/UserController.java

```diff
@@ -58,13 +67,20 @@ public UserGetDTO createUser(@RequestBody UserPostDTO userPostDTO) {
     // convert internal representation of user back to API
     UserGetDTO userGetDTO = DTOMapper.INSTANCE.convertEntityToUserGetDTO(createdUser);
 
-    return userGetDTO;
+    // create HttpHeaders object, add token in response header and make it accessible to the client
+    HttpHeaders headers = new org.springframework.http.HttpHeaders();
+    headers.add("X-Token", createdUser.getToken());
+    List<String> customHeaders = new ArrayList<String>();
+    customHeaders.add("X-Token");
+    headers.setAccessControlExposeHeaders(customHeaders);
+
+    return ResponseEntity.status(HttpStatus.CREATED).headers(headers).body(userGetDTO);
   }
 
-  @PutMapping("users/{username}/login")
+  @PostMapping("users/{username}/login")
```

Comment:
I think it would help if we shortly mention the Http calls with the corresponding code always next to it, similar to here. Helps checking the REST table :)

Reply:
But we can add this later on